Telemetry/Telemetry.cpp

```diff
@@ -0,0 +1,173 @@
+        {
+            LOGINFOMETHOD();
+
+            JsonArray profiles;
+            getProfiles(profiles);
+            response["reportProfiles"] = profiles;
+
+            returnResponse(true);
+        }
+
+        uint32_t Telemetry::setReportProfileStatus(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+
+            if (parameters.HasLabel("reportProfile") && parameters.HasLabel("status"))
+            {
+                string status;
+                getStringParameter("status", status);
+
+                if (status != "STARTED" && status != "COMPLETE")
+                {
+                    LOGERR("Only the 'STARTED' or 'COMPLETE' status is allowed");
+                    returnResponse(false);
+                }
+
+                string reportProfile;
+                getStringParameter("reportProfile", reportProfile);
+
+                JsonArray profiles;
+                getProfiles(profiles);
+
+                JsonArray::Iterator it = profiles.Elements();
+                while(it.Next())
+                {
+                    std::string profile = it.Current().String();
+                    if (profile == reportProfile)
+                    {
+                        std::string rfc = RFC_REPORT_PROFILE_ENABLE;
+                        rfc += profile;
+                        rfc += "Report.Enable";
+
+                        WDMP_STATUS wdmpStatus = setRFCParameter(RFC_CALLERID, rfc.c_str(), status == "COMPLETE" ? "true" : "false", WDMP_BOOLEAN);
+                        if (WDMP_SUCCESS != wdmpStatus)
+                        {
+                            LOGERR("Failed to set %s: %d", rfc.c_str(), wdmpStatus);
+                            returnResponse(false);
+                        }
+
+                        returnResponse(true);
+                    }
+                }
+            }
+            else
+            {
+                LOGERR("No 'status' parameter");
+                returnResponse(false);
+            }
+
+            returnResponse(true);
+        }
+
+        uint32_t Telemetry::logApplicationEvent(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+
+            if (parameters.HasLabel("eventName") && parameters.HasLabel("eventValue"))
+            {
+                string eventName;
+                getStringParameter("eventName", eventName);
+
+                string eventValue;
+                getStringParameter("eventValue", eventValue);
+
+                LOGT2((char *)eventName.c_str(), (char *)eventValue.c_str());
+            }
+            else
+            {
+                LOGERR("No 'eventName' or 'eventValue' parameter");
+                returnResponse(false);
+            }
+
+            returnResponse(true);
+        }
+
+        void Telemetry::getProfiles(JsonArray &profiles)
+        {
+            profiles.Clear();
+
+            DIR *d = opendir("/opt/.t2persistentfolder/");
```

Comment:
Where are we reading the default FTUE report profile?

Reply:
Looks like there is no such thing as default profile. Should I hardcode "FTUE" in the case of empty /opt/.t2persistentfolder/ ?

Reply:
No, we shouldn't hardcode. Let's check with Mahee on where to read the default report profile. Approving these current changes.